src/main/java/io/github/llamarama/team/common/entity/caravantrader/CaravanTraderEntity.java

```diff
@@ -174,4 +175,13 @@ protected SoundEvent getSplashSound() {
         return SoundEvents.ENTITY_GENERIC_SPLASH;
     }
 
+    @Override
+    public void tickMovement() {
+        if (lifespan > 0) {
```

Comment:
Also this will make *all* traders despawn. However, we only need the ones the player *hasn't* traded with to do so.